Momento/MomentoSigner.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.IdentityModel.Tokens.Jwt;
+using Microsoft.IdentityModel.Tokens;
+using MomentoSdk.Exceptions;
+
+namespace MomentoSdk
+{
+    public class MomentoSigner
+    {
+
+        /// <summary>
+        /// Create a pre-signed HTTPS URL to retrieve an object.
+        /// </summary>
+        /// <param name="hostname">Hostname of the SimpleCacheService. Use the value returned from CreateSigningKey's response.</param>
+        /// <param name="jwkJsonString">JWK in JSON string. Use the value returned from CreateSigningKey's response.</param>
+        /// <param name="cacheName">The name of the Cache.</param>
+        /// <param name="cacheKey">The key of the Object.</param>
+        /// <param name="expiryEpochSeconds">The timestamp that the pre-signed URL is valid until.</param>
+        /// <returns></returns>
+        public string CreatePresignedUrlForGet(string hostname, string jwkJsonString, string cacheName, string cacheKey, uint expiryEpochSeconds)
+        {
+            var payload = CommonJwtBody(cacheName, cacheKey, expiryEpochSeconds);
+            payload.Add("method", new string[] { "get" });
+
+            var encodedJwt = CreateJwtToken(jwkJsonString, payload);
+
+            return $"https://{hostname}/cache/get/{cacheName}/{cacheKey}?token={encodedJwt}";
+        }
+
+        /// <summary>
+        /// Create a pre-signed HTTPS URL to upload an object.
+        /// </summary>
+        /// <param name="hostname">Hostname of the SimpleCacheService. Use the value returned from CreateSigningKey's response.</param>
+        /// <param name="jwkJsonString">JWK in JSON string. Use the value returned from CreateSigningKey's response.</param>
+        /// <param name="cacheName">The name of the Cache.</param>
+        /// <param name="cacheKey">The key of the Object.</param>
+        /// <param name="expiryEpochSeconds">The timestamp that the pre-signed URL is valid until.</param>
+        /// <param name="ttlSeconds">The timestamp that the object is valid until.</param>
```

Comment:
We should probably have this as the duration in seconds that the item will be stored in cache so that our grpc requests are consistent. Also, it needs to be parsed to duration in milli and given as a query param https://github.com/momentohq/client_protos/blob/main/proto/httpcache.proto#L21
I see that on the design doc there's ttl, cache and key in the cliam. I don't think we need them actually.  Mr2 is getting them from the grpc request already (mapped from name and key in the path, and the query param ttl)

Reply:
so I think cache, key, and TTL are included in the claim for authorization purposes. Any client can try to access a different cache, key, or change their own TTL in the request/query param, but if it's included in the claim MR2 can verify (via the signature) that the URL was generated for that particular resource and the client isn't trying to access things outside its permitted scope.

Reply:
Makes sense! Thanks!

Reply:
+1 to Daniela said about authorization.

Also I'm going to keep this as `ttlSeconds` to be consistent with the existing `SimpleCacheClient` in this SDK.

Reply:
Yup I understand the purpose of having these fields in the claim now. 
Regarding `ttlSeconds`, I was referring to the doc. It's not `The timestamp that the object is valid until`, but the ttl duration.  To be consistent we can use `Time to Live for the item in Cache.`?
We should still parse `ttlSeconds` to milli twhen adding ttl to payload, so that it matches our proto definition. Just like the existing SDK behavior for grpc get requests https://github.com/momentohq/client-sdk-csharp/blob/main/Momento/ScsDataClient.cs#L166.

Reply:
Oh right! Good catch on both of those.

Reply:
Fixed.